assets/src/edit-story/elements/shape/layer.js

```diff
@@ -15,12 +15,76 @@
  */
 
 /**
- * WordPress dependencies
+ * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import styled from 'styled-components';
+import { useMemo } from 'react';
 
-function ShapeLayerContent() {
-  return __('Shape', 'web-stories');
+/**
+ * Internal dependencies
+ */
+import { getMaskByType } from '../../masks';
+import { elementWithFillColor } from '../shared';
+import StoryPropTypes from '../../types';
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+
+const ShapePreview = styled.div`
+  position: relative;
+  display: flex;
+  justify-content: center;
+  margin-right: 8px;
+`;
+
+const ShapeSVG = styled.svg`
+  ${({ isTooBright }) =>
+    isTooBright &&
+    `
+    /* Using filter rather than box-shadow to correctly follow
+     * outlines in semi-transparent images like gif and png.
+     */
+    filter: drop-shadow( 0 0 5px rgba(0, 0, 0, 0.5) );
+  `}
+`;
+
+const ShapePath = styled.path`
+  ${elementWithFillColor}
+`;
+
+function ShapeLayerContent({ element: { mask, backgroundColor } }) {
+  const maskDef = getMaskByType(mask.type);
+
+  const isTooBright = useMemo(() => {
+    const darkestDimension = Math.max.apply(
+      null,
+      Object.values(backgroundColor?.color || {})
+    );
+    return darkestDimension >= 230;
```

Comment:
Maybe define it as `const TOO_BRIGHT = 230` like in `visibleImage.js`? Otherwise we might miss one instance in case we'd have to change this value.

Or perhaps extract this function so it can be shared